src/main/java/core/basesyntax/service/impl/DeepCopyImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.FruitTransaction;
+import core.basesyntax.service.DeepCopy;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DeepCopyImpl implements DeepCopy {
+    @Override
+    public List<FruitTransaction> getDeepCopy(List<FruitTransaction> list) {
+        List<FruitTransaction> copyList = new ArrayList<>(list.size());
+        for (FruitTransaction fruit : list) {
+            copyList.add(new FruitTransaction(fruit.getType(),
+                    fruit.getFruit(), fruit.getQuantity()));
+        }
+        return copyList;
```

Comment:
The implementation of the deep copy method is correct, but it relies on the constructor of `FruitTransaction` to copy the object. Ensure that the constructor itself performs a deep copy of any mutable objects to avoid shared references.